support remote use

p-wall opened the pull request:
if HOSTNAME is set (should be, on both mac and linux), that will be used for urls instead of localhost listen on all interfaces to allow remote access, for example for kibana

p-wall commented:
an alternative could be to just use `Socket.gethostname` always, which i would prefer but you might not
this works with 
```bash
export HOSTNAME
k kibana some-cluster
```

or 

```bash
HOSTNAME=$HOSTNAME k kibana some-cluster
```

we could also use a dedicated env, like `DEV_HOST`

dbackeus commented:
I have security concerns about this one. If I understand the implications of `--bind 0.0.0.0` correectly, if I would run eg. `k pg:proxy` I would now expose all postgres databases on the network, given someone could guess the name of the application which could be trivial (eg `mynewsdesk` for our main production database).

p-wall commented:
right, that wouldn't affect us since we don't expose our devboxes to a network without a firewall, perhaps selectively enable it depending on an env?

p-wall commented:
gated it behind `K_LISTEN_ON_ALL_INTERFACES`

dbackeus commented on k:
```diff
@@ -5,11 +5,17 @@
 require "yaml"
 require "tmpdir"
 require "uri"
+require "socket"
```
I'm trying to be hardcore about avoiding `require` on the main path, to keep `k` execution as fast as possible. Maybe relying on `HOSTNAME` env var is good enough at this stage?

p-wall replied:
how about just putting the require in an `if`, depending on the `K_LISTEN_ON_ALL_INTERFACES` env?

dbackeus replied:
Can be OK

dbackeus commented on k:
```diff
@@ -5,11 +5,17 @@
 require "uri"
+require "socket"
+
+LISTEN_ON_ALL_INTERFACES = %w[1 true t yes y on].include?(ENV.fetch("K_LISTEN_ON_ALL_INTERFACES", "").downcase)
```
Let's reduce this to only look for `true` (can also change the comment explaining the ENV var at the top of the file).

Eg.

```suggestion
LISTEN_ON_ALL_INTERFACES = ENV["K_LISTEN_ON_ALL_INTERFACES"] == "true"
```

p-wall commented:
moved logic into `def kibana`, where `socket` was already required, since that was the only affected function in `k`
changed logic to be simple if/else instead of repeated ternaries
changed accepted values to only "true"

dbackeus commented on k:
```diff
@@ -594,12 +595,19 @@ def kibana
   # kibana.k8s.elastic.co/name=mynewsdesk-funnel
 
   require "socket"
```
Looks like this should still be moved into the conditional.

p-wall replied:
this was here before my changes, it's needed for `TCPSocket` below

dbackeus replied:
I didn't realize this was inside of the kibana method. Lack of context on github 😅

dbackeus commented:
@p-wall did you forget to push your latest changes? The diff doesn't seem to reflect your comment.

p-wall commented:
> @p-wall did you forget to push your latest changes? The diff doesn't seem to reflect your comment.

@dbackeus i can see my latest changes in https://github.com/reclaim-the-stack/k/pull/18/files
